### PR TITLE
Tolerate missing field in node metadata platform metadata

### PR DESCRIPTION
**Pull request by bianpengyuan**
fix https://github.com/istio/istio/issues/18614

### CODE REVIEW

File: extensions/stackdriver/log/logger.cc [hunk 1]
```diff
@@ -44,15 +44,18 @@ Logger::Logger(const ::wasm::common::NodeInfo& local_node_info,
...
   // Set log names.
-  const auto& project_id =
-      local_node_info.platform_metadata().at(Common::kGCPProjectKey);
+  const auto& platform_metadata = local_node_info.platform_metadata();
+  auto iter = platform_metadata.find(Common::kGCPProjectKey);
+  std::string project_id = "";
+  if (iter != platform_metadata.end()) {
+    project_id = iter->second;
+  }
   log_entries_request_->set_log_name("projects/" + project_id + "/logs/" +
```
Comment (kyessenov) [inline extensions/stackdriver/log/logger.cc]: maybe we should skip setting log name if project ID is not found? or is there a better default?

Comment (bianpengyuan) [reply extensions/stackdriver/log/logger.cc]: There is no good fallback, we should just use the validation API to reject the listener config. I added a TODO for it, will do it in another PR.

File: extensions/stackdriver/edges/edge_reporter.cc [hunk 2]
```diff
@@ -47,10 +47,15 @@ void instanceFromMetadata(const ::wasm::common::NodeInfo& node_info,
   absl::StrAppend(instance->mutable_uid(), "kubernetes://", node_info.name(),
                   ".", node_info.namespace_());
   // TODO(douglas-reid): support more than just GCP ?
-  instance->set_location(
-      node_info.platform_metadata().at(Common::kGCPLocationKey));
-  instance->set_cluster_name(
-      node_info.platform_metadata().at(Common::kGCPClusterNameKey));
+  const auto& platform_metadata = node_info.platform_metadata();
+  auto iter = platform_metadata.find(Common::kGCPLocationKey);
```
Comment (kyessenov) [inline extensions/stackdriver/edges/edge_reporter.cc]: nit: const to force const iterator

Comment (bianpengyuan) [reply extensions/stackdriver/edges/edge_reporter.cc]: Done.

**Pull request merged**